src/Http/Routing/src/Matching/CandidateState.cs

```diff
@@ -50,6 +50,6 @@ internal CandidateState(Endpoint endpoint, RouteValueDictionary values, int scor
         /// Gets <see cref="RouteValueDictionary"/> associated with the
         /// <see cref="Http.Endpoint"/> and the current request.
         /// </summary>
-        public RouteValueDictionary Values { get; internal set; }
+        public RouteValueDictionary Values { get; set; }
```

Comment:
@dougbu @JunTaoLuo I had to change this setter to not be internal, else `/t:GenerateReferenceSources` would undo the change to https://github.com/aspnet/AspNetCore/pull/17311/commits/49379ed50480afa0b0009960c8ecbcb7dace50a6#diff-24bb8cbda21395fb70004db888e179d5R520, causing a build break in the following test:

https://github.com/aspnet/AspNetCore/blob/e0ca7facf08f4bb978adc0c55d48f78d82ffdac8/src/Http/Routing/test/UnitTests/Matching/DefaultEndpointSelectorTest.cs#L42

Is this the right fix?

Reply:
Specifically:

> DefaultEndpointSelectorTest.cs(42,13): error CS0200: Property or indexer 'CandidateState.Values' cannot be assigned to -- it is read only [E:\code\aspnet\aspnetcore\src\Http\Routing\test\UnitTests\Microsoft.AspNetCore.Routing.Tests.csproj]

Reply:
Tagging @rynowak.  I don't think that's the right fix.  If we can't do this another way, I would suggest instead removing the setter, and making an internal `SetValues()` method :(

Reply:
@JunTaoLuo has done this in the past by excluding the relevant API in GenAPI, and adding the setter manually to the `manual.cs` file - I did that here: https://github.com/aspnet/AspNetCore/pull/17311/commits/b43e386df9d0968834efad3796bd9ab6f13a19a4

Reply:
Perfect - thanks.  I didn't know we had GenAPI exclusions set up to support cases like this.

Reply:
Yup, I ran into similar issues in 3.0.1 and this seems like the most convenient and reliable way to resolve it.